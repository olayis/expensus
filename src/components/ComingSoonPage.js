import React from 'react';
import { Link } from 'react-router-dom';
import programming from '../../public/img/programming.svg';

const ComingSoonPage = () => (
  <main className='main-content'>
    <div className='content-container'>
      <div className='error'>
        <img
          src={programming}
          alt='Programmer'
          className='error__illustration'
          width='1075'
          height='585'
        />
        <h2 className='error__text'>This page is coming soon.</h2>
        <div className='error__explanation'>
          Continue managing your expenses.
          <span className='error__action'>
            <Link className='btn btn--try-again' to='/'>
              Go Home
            </Link>
          </span>
        </div>
      </div>
    </div>
  </main>
);

export default ComingSoonPage;
