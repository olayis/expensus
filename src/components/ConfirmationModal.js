import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => {
  if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

  return (
    <div>
      <Modal
        contentLabel='Are you sure?'
        isOpen={!!props.deleteSelected}
        onRequestClose={props.stopDeleteDataSelected}
        closeTimeoutMS={200}
        className='modal'
        aria={{
          labelledby: 'heading',
          describedby: 'full_description',
        }}
      >
        <h3 id='heading' className='modal__title'>
          Delete Data
        </h3>
        <p id='full_description' className='modal__body'>
          Are you sure you want to delete all of your data on Expensus?
        </p>
        <button className='modal__btn--delete' onClick={props.handleDeleteData}>
          Delete
        </button>
        <button
          className='modal__btn--cancel'
          onClick={props.stopDeleteDataSelected}
        >
          Cancel
        </button>
      </Modal>
    </div>
  );
};

export default ConfirmationModal;
