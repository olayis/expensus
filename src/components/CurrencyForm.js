import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';

export default class CurrencyForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currency: props.currency,
      currencyError: '',
    };
  }

  onCurrencyChange = (e) => {
    const currency = e.target.value;
    this.setState(() => ({ currency }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.currency) {
      this.setState(() => ({
        currencyError: 'Please select a valid currency',
      }));
    } else {
      this.setState(() => ({ currencyError: '' }));
      this.props.onSubmit({
        currency: this.state.currency.trim(),
      });
    }
  };

  render() {
    return (
      <form className='input-field' onSubmit={this.onSubmit}>
        <div className='input-field__wrapper'>
          <select
            className='dropdown dropdown--currency'
            value={this.state.currency}
            onChange={this.onCurrencyChange}
            aria-label='Select Currency'
            name='currency'
          >
            <option value=''>-- Select Currency --</option>
            <option value='Baht'>Baht (฿)</option>
            <option value='Danish Krone'>Danish Krone (DKK)</option>
            <option value='Dollar'>Dollar ($)</option>
            <option value='Dong'>Dong (₫)</option>
            <option value='Euro'>Euro (€)</option>
            <option value='Forint'>Forint (Ft)</option>
            <option value='Hryvnia'>Hryvnia (₴)</option>
            <option value='Koruna'>Koruna (Kč)</option>
            <option value='Lira'>Lira (₺)</option>
            <option value='Lev'>Lev (лв)</option>
            <option value='Naira'>Naira (₦)</option>
            <option value='Norwegian krone'>Norwegian krone (kr)</option>
            <option value='Pound Sterling'>Pound Sterling (£)</option>
            <option value='Rand'>Rand (R)</option>
            <option value='Real'>Real (R$)</option>
            <option value='Ruble'>Ruble (₽)</option>
            <option value='Swiss Franc'>Swiss Franc (CHF)</option>
            <option value='Yen'>Yen (¥)</option>
            <option value='Yuan'>Yuan (¥)</option>
            <option value='Zloty'>Złoty (PLN)</option>
          </select>
          {this.state.currencyError && (
            <span className='input-field__error-msg'>
              <FontAwesomeIcon className='fa-icon' icon={faExclamationCircle} />
              {this.state.currencyError}
            </span>
          )}
        </div>
        <button className='btn btn-currency'>Change Currency</button>
      </form>
    );
  }
}
