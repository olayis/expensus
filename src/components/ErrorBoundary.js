import React from 'react';
import store from '../store/configureStore';
import ErrorPage from './ErrorPage';
import startLogError from '../actions/logError';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    process.env.NODE_ENV === 'Production'
      ? store.dispatch(startLogError(error, errorInfo))
      : '';
  }

  render() {
    if (this.state.hasError) {
      return <ErrorPage />;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
