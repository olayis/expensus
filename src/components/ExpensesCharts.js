import React from 'react';
import { connect } from 'react-redux';
import { Chart } from 'react-google-charts';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChartBar } from '@fortawesome/free-regular-svg-icons';
import selectExpenses from '../selectors/expenses';
import charts from '../../public/img/charts.svg';
import spinner from '../../public/img/spinner.gif';
import categoryCountAmountPercentage from '../statistics/categoryCountAmount';
import dailyAmount from '../statistics/dailyAmount';
import dailyCount from '../statistics/dailyCount';
import expenseAmount from '../statistics/descriptionAmount';

export const ExpensesCharts = (props) => {
  const pieChartData = expenseAmount(props.expenses);
  const barChartData = dailyAmount(props.expenses);
  const lineChartData = dailyCount(props.expenses);
  const bubbleChartData = categoryCountAmountPercentage(props.expenses);

  return (
    <section className='content-container'>
      <div className='component-container component-container--chart'>
        <h2 className='heading-secondary'>
          <FontAwesomeIcon icon={faChartBar} /> Expenses Charts
        </h2>
        {props.expenses.length === 0 ? (
          <div className='no-expense'>
            <img
              src={charts}
              alt='A man looking at charts'
              className='no-expense__illustration'
              width='773'
              height='532'
            />
            <p>Start adding Expenses to show Charts.</p>
          </div>
        ) : (
          <div>
            <div className='row'>
              <div className='col-1-of-2'>
                <div className='expense-chart'>
                  <div className='expense-chart__title'>
                    Expenses (By Percentage)
                  </div>
                  <Chart
                    className='expense-chart__chart'
                    width={'100%'}
                    height={'30rem'}
                    chartType='PieChart'
                    loader={
                      <div>
                        <img src={spinner} alt='' width='200' height='200' />
                        <p>Loading Pie Chart...</p>
                      </div>
                    }
                    data={pieChartData}
                    rootProps={{ 'data-test-id': '1' }}
                  />
                </div>
              </div>
              <div className='col-1-of-2'>
                <div className='expense-chart'>
                  <div className='expense-chart__title'>Expenses Pie</div>
                  <Chart
                    className='expense-chart__chart'
                    width={'100%'}
                    height={'30rem'}
                    chartType='PieChart'
                    loader={
                      <div>
                        <img src={spinner} alt='' width='200' height='200' />
                        <p>Loading Chart...</p>
                      </div>
                    }
                    data={pieChartData}
                    options={{
                      pieHole: 0.5,
                    }}
                    rootProps={{ 'data-test-id': '2' }}
                  />
                </div>
              </div>
            </div>
            <div className='row'>
              <div className='col-1-of-2'>
                <div className='expense-chart'>
                  <div className='expense-chart__title'>
                    Daily Amount Of Expenses
                  </div>
                  <Chart
                    className='expense-chart__chart'
                    width={'100%'}
                    height={'30rem'}
                    chartType='Bar'
                    loader={
                      <div>
                        <img src={spinner} alt='' width='200' height='200' />
                        <p>Loading Bar Chart...</p>
                      </div>
                    }
                    data={barChartData}
                    rootProps={{ 'data-testid': '3' }}
                  />
                </div>
              </div>
              <div className='col-1-of-2'>
                <div className='expense-chart'>
                  <div className='expense-chart__title'>
                    Number of Daily Expenses
                  </div>
                  <Chart
                    className='expense-chart__chart'
                    width={'100%'}
                    height={'30rem'}
                    chartType='LineChart'
                    loader={
                      <div>
                        <img src={spinner} alt='' width='200' height='200' />
                        <p>Loading Line Chart...</p>
                      </div>
                    }
                    data={lineChartData}
                    options={{
                      hAxis: {
                        title: 'Date',
                      },
                      vAxis: {
                        title: 'Number',
                      },
                    }}
                    rootProps={{ 'data-testid': '4' }}
                  />
                </div>
              </div>
            </div>
            <div className='row'>
              <div className='expense-chart'>
                <div className='expense-chart__title'>Expenses Categories</div>
                <Chart
                  className='expense-chart__chart'
                  width={'100%'}
                  height={'50rem'}
                  chartType='BubbleChart'
                  loader={
                    <div>
                      <img src={spinner} alt='' width='200' height='200' />
                      <p>Loading Bubble Chart...</p>
                    </div>
                  }
                  data={bubbleChartData}
                  options={{
                    hAxis: { title: 'Category' },
                    vAxis: { title: 'Number' },
                  }}
                  rootProps={{ 'data-testid': '5' }}
                />
              </div>
            </div>
          </div>
        )}
      </div>
    </section>
  );
};

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpensesCharts);
