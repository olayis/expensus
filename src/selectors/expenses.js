// Get Visible expenses
import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter((expense) => {
      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdAtMoment, 'day')
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdAtMoment, 'day')
        : true;
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());

      return textMatch && startDateMatch && endDateMatch;
    })
    .sort((a, b) => {
      switch (sortBy) {
        case 'date(recent)':
          return a.createdAt < b.createdAt ? 1 : -1;

        case 'date(oldest)':
          return a.createdAt > b.createdAt ? 1 : -1;

        case 'amount(highest)':
          return a.amount < b.amount ? 1 : -1;

        case 'amount(lowest)':
          return a.amount > b.amount ? 1 : -1;

        case 'description(a_to_z)':
          return a.description.localeCompare(b.description);

        case 'description(z_to_a)':
          return b.description.localeCompare(a.description);

        case 'category':
          return a.category.localeCompare(b.category);
      }
    });
};
