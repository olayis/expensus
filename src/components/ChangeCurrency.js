import React from 'react';
import { connect } from 'react-redux';
import { startChangeLocale } from '../actions/locale';
import CurrencyForm from './CurrencyForm';

export class ChangeCurrency extends React.Component {
  onSubmit = ({ currency }) => {
    this.props.startChangeLocale({ currency });
    this.props.history.push('/');
    this.props.history.go(0);
  };

  goBack = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='component-container component-container--currency'>
            <CurrencyForm
              onSubmit={this.onSubmit}
              currency={this.props.currency}
              goBack={this.goBack}
            />
          </div>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state) => ({
  currency: state.locale ? state.locale.currency : '',
});

const mapDispatchToProps = (dispatch) => ({
  startChangeLocale: (currency) => dispatch(startChangeLocale(currency)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCurrency);
