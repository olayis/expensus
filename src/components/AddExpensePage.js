import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div className='main-content'>
        <div className='content-container'>
          <div className='add-expense-container'>
            <ExpenseForm onSubmit={this.onSubmit} />
          </div>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
