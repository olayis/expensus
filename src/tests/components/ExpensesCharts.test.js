import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesCharts } from '../../components/ExpensesCharts';

test('should render ExpensesChart with expenses data', () => {
  const wrapper = shallow(<ExpensesCharts expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesCharts with empty message', () => {
  const wrapper = shallow(<ExpensesCharts expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
