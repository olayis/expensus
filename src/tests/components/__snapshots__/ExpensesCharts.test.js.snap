// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesChart with expenses data 1`] = `
<div
  className="content-container"
>
  <div
    className="expenses-chart-container"
  >
    <h2
      className="heading-secondary"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f080",
              "M396.8 352h22.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-192 0h22.4c6.4 0 12.8-6.4 12.8-12.8V140.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h22.4c6.4 0 12.8-6.4 12.8-12.8V204.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zM496 400H48V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16zm-387.2-48h22.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8z",
            ],
            "iconName": "chart-bar",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       Expenses Charts
    </h2>
    <div>
      <div
        className="row"
      >
        <div
          className="col-1-of-2"
        >
          <div
            className="expense-chart"
          >
            <div
              className="expense-chart__title"
            >
              Expenses (By Percentage)
            </div>
            <Chart
              chartActions={null}
              chartEvents={null}
              chartType="PieChart"
              chartWrapperParams={Object {}}
              className="expense-chart__chart"
              columns={null}
              controls={null}
              data={
                Array [
                  Array [
                    "Title",
                    "Amount",
                  ],
                  Array [
                    "Data",
                    20,
                  ],
                  Array [
                    "Home Rentage",
                    350,
                  ],
                  Array [
                    "Room Repainting",
                    80,
                  ],
                  Array [
                    "Water Pipes Repair",
                    10,
                  ],
                ]
              }
              diffdata={null}
              formatters={null}
              getChartEditor={null}
              getChartWrapper={[Function]}
              graphID={null}
              graph_id={null}
              height="30rem"
              legendToggle={false}
              legend_toggle={false}
              loader={
                <div>
                  <img
                    src="test-file-stub"
                  />
                  <p>
                    Loading Pie Chart...
                  </p>
                </div>
              }
              options={
                Object {
                  "colors": null,
                }
              }
              render={null}
              rootProps={
                Object {
                  "data-test-id": "1",
                }
              }
              rows={null}
              spreadSheetQueryParameters={
                Object {
                  "gid": 1,
                  "headers": 1,
                }
              }
              spreadSheetUrl={null}
              style={Object {}}
              toolbarID={null}
              toolbarItems={null}
              width="100%"
            />
          </div>
        </div>
        <div
          className="col-1-of-2"
        >
          <div
            className="expense-chart"
          >
            <div
              className="expense-chart__title"
            >
              Expenses Pie
            </div>
            <Chart
              chartActions={null}
              chartEvents={null}
              chartType="PieChart"
              chartWrapperParams={Object {}}
              className="expense-chart__chart"
              columns={null}
              controls={null}
              data={
                Array [
                  Array [
                    "Title",
                    "Amount",
                  ],
                  Array [
                    "Data",
                    20,
                  ],
                  Array [
                    "Home Rentage",
                    350,
                  ],
                  Array [
                    "Room Repainting",
                    80,
                  ],
                  Array [
                    "Water Pipes Repair",
                    10,
                  ],
                ]
              }
              diffdata={null}
              formatters={null}
              getChartEditor={null}
              getChartWrapper={[Function]}
              graphID={null}
              graph_id={null}
              height="30rem"
              legendToggle={false}
              legend_toggle={false}
              loader={
                <div>
                  <img
                    src="test-file-stub"
                  />
                  <p>
                    Loading Chart...
                  </p>
                </div>
              }
              options={
                Object {
                  "pieHole": 0.5,
                }
              }
              render={null}
              rootProps={
                Object {
                  "data-test-id": "2",
                }
              }
              rows={null}
              spreadSheetQueryParameters={
                Object {
                  "gid": 1,
                  "headers": 1,
                }
              }
              spreadSheetUrl={null}
              style={Object {}}
              toolbarID={null}
              toolbarItems={null}
              width="100%"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1-of-2"
        >
          <div
            className="expense-chart"
          >
            <div
              className="expense-chart__title"
            >
              Daily Amount Of Expenses
            </div>
            <Chart
              chartActions={null}
              chartEvents={null}
              chartType="Bar"
              chartWrapperParams={Object {}}
              className="expense-chart__chart"
              columns={null}
              controls={null}
              data={
                Array [
                  Array [
                    "Date",
                    "Amount",
                  ],
                  Array [
                    "01/01",
                    20,
                  ],
                  Array [
                    "27/12",
                    350,
                  ],
                  Array [
                    "13/01",
                    80,
                  ],
                  Array [
                    "11/01",
                    10,
                  ],
                ]
              }
              diffdata={null}
              formatters={null}
              getChartEditor={null}
              getChartWrapper={[Function]}
              graphID={null}
              graph_id={null}
              height="30rem"
              legendToggle={false}
              legend_toggle={false}
              loader={
                <div>
                  <img
                    src="test-file-stub"
                  />
                  <p>
                    Loading Bar Chart...
                  </p>
                </div>
              }
              options={
                Object {
                  "colors": null,
                }
              }
              render={null}
              rootProps={
                Object {
                  "data-testid": "3",
                }
              }
              rows={null}
              spreadSheetQueryParameters={
                Object {
                  "gid": 1,
                  "headers": 1,
                }
              }
              spreadSheetUrl={null}
              style={Object {}}
              toolbarID={null}
              toolbarItems={null}
              width="100%"
            />
          </div>
        </div>
        <div
          className="col-1-of-2"
        >
          <div
            className="expense-chart"
          >
            <div
              className="expense-chart__title"
            >
              Number of Daily Expenses
            </div>
            <Chart
              chartActions={null}
              chartEvents={null}
              chartType="LineChart"
              chartWrapperParams={Object {}}
              className="expense-chart__chart"
              columns={null}
              controls={null}
              data={
                Array [
                  Array [
                    "Date",
                    "Number Of Expenses",
                  ],
                  Array [
                    "01/01",
                    1,
                  ],
                  Array [
                    "27/12",
                    1,
                  ],
                  Array [
                    "13/01",
                    1,
                  ],
                  Array [
                    "11/01",
                    1,
                  ],
                ]
              }
              diffdata={null}
              formatters={null}
              getChartEditor={null}
              getChartWrapper={[Function]}
              graphID={null}
              graph_id={null}
              height="30rem"
              legendToggle={false}
              legend_toggle={false}
              loader={
                <div>
                  <img
                    src="test-file-stub"
                  />
                  <p>
                    Loading Line Chart...
                  </p>
                </div>
              }
              options={
                Object {
                  "hAxis": Object {
                    "title": "Date",
                  },
                  "vAxis": Object {
                    "title": "Number",
                  },
                }
              }
              render={null}
              rootProps={
                Object {
                  "data-testid": "4",
                }
              }
              rows={null}
              spreadSheetQueryParameters={
                Object {
                  "gid": 1,
                  "headers": 1,
                }
              }
              spreadSheetUrl={null}
              style={Object {}}
              toolbarID={null}
              toolbarItems={null}
              width="100%"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="expense-chart"
        >
          <div
            className="expense-chart__title"
          >
            Expenses Categories
          </div>
          <Chart
            chartActions={null}
            chartEvents={null}
            chartType="BubbleChart"
            chartWrapperParams={Object {}}
            className="expense-chart__chart"
            columns={null}
            controls={null}
            data={
              Array [
                Array [
                  "ID",
                  "Total",
                  "Number",
                  "Category",
                  "Percentage",
                ],
                Array [
                  "Bills & Utilities",
                  20,
                  1,
                  "Bills & Utilities",
                  4.3478260869565215,
                ],
                Array [
                  "Rent",
                  350,
                  1,
                  "Rent",
                  76.08695652173914,
                ],
                Array [
                  "Maintenance",
                  90,
                  2,
                  "Maintenance",
                  19.565217391304348,
                ],
              ]
            }
            diffdata={null}
            formatters={null}
            getChartEditor={null}
            getChartWrapper={[Function]}
            graphID={null}
            graph_id={null}
            height="50rem"
            legendToggle={false}
            legend_toggle={false}
            loader={
              <div>
                <img
                  src="test-file-stub"
                />
                <p>
                  Loading Bubble Chart...
                </p>
              </div>
            }
            options={
              Object {
                "hAxis": Object {
                  "title": "Category",
                },
                "vAxis": Object {
                  "title": "Number",
                },
              }
            }
            render={null}
            rootProps={
              Object {
                "data-testid": "5",
              }
            }
            rows={null}
            spreadSheetQueryParameters={
              Object {
                "gid": 1,
                "headers": 1,
              }
            }
            spreadSheetUrl={null}
            style={Object {}}
            toolbarID={null}
            toolbarItems={null}
            width="100%"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesCharts with empty message 1`] = `
<div
  className="content-container"
>
  <div
    className="expenses-chart-container"
  >
    <h2
      className="heading-secondary"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f080",
              "M396.8 352h22.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-192 0h22.4c6.4 0 12.8-6.4 12.8-12.8V140.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h22.4c6.4 0 12.8-6.4 12.8-12.8V204.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zM496 400H48V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16zm-387.2-48h22.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8z",
            ],
            "iconName": "chart-bar",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       Expenses Charts
    </h2>
    <div
      className="no-expense"
    >
      <img
        alt="A man looking at charts"
        className="no-expense__illustration"
        src="SvgrURL"
      />
      <p>
        Start adding Expenses to show Charts.
      </p>
    </div>
  </div>
</div>
`;
